x/stake/keeper/validator.go

```diff
@@ -210,19 +210,26 @@ func (k Keeper) UpdateValidator(ctx sdk.Context, validator types.Validator) type
 	cliffPower := k.GetCliffValidatorPower(ctx)
 
 	switch {
-	// if already bonded and power increasing only need to update tendermint
+	// if the validator is already bonded and the power is increasing, we need
+	// perform the following:
+	// a) update Tendermint
+	// b) check if the cliff validator needs to be updated
 	case powerIncreasing && !validator.Revoked &&
 		(oldFound && oldValidator.Status == sdk.Bonded):
 
 		bz := k.cdc.MustMarshalBinary(validator.ABCIValidator())
 		store.Set(GetTendermintUpdatesKey(validator.Owner), bz)
 
+		if cliffPower != nil {
+			k.updateCliffValidator(ctx, validator)
```

Comment:
from the godoc of k.updateCliffVlaidator, it seems that we should have the cliffPower != nil check in there?

Reply:
Not sure I totally agree, I think have the explicit check here makes the context more clear opposed to always calling `updateCliffValidator`.